ext/spl/php_spl.c

```diff
@@ -508,86 +508,46 @@ PHP_FUNCTION(spl_autoload_call)
 PHP_FUNCTION(spl_autoload_register)
 {
 	zend_string *func_name;
-	char *error = NULL;
 	zend_string *lc_name;
-	zval *zcallable = NULL;
 	zend_bool do_throw = 1;
 	zend_bool prepend  = 0;
 	zend_function *spl_func_ptr;
 	autoload_func_info alfi;
 	zend_object *obj_ptr;
+	zend_fcall_info fci = {0};
 	zend_fcall_info_cache fcc;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|z!bb", &zcallable, &do_throw, &prepend) == FAILURE) {
-		RETURN_THROWS();
+	ZEND_PARSE_PARAMETERS_START(0, 3)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_FUNC_OR_NULL(fci, fcc)
+		Z_PARAM_BOOL(do_throw)
+		Z_PARAM_BOOL(prepend)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (!do_throw) {
+		php_error_docref(NULL, E_WARNING, "spl_autoload_register will always throw, "
+			"the second argument has been ignored");
```

Comment:
Maybe follow the `Argument #2 ($do_throw) has been ignored, spl_autoload_register() will always throw` style here? I'd also downgrade this to E_NOTICE. By the time we reach this code the exception has already been thrown anyway, so this is purely informational.

Reply:
@kocsismate Is there some existing function for this, for the non-exception case? Or do we just write out the error message explicitly?

Reply:
No, we've only added utility functions to handle exceptions yet. So yes, explicitly writing this out is the only solution for now.